buildSrc/src/main/java/org/opensearch/gradle/DistributionDownloadPlugin.java

```diff
@@ -234,7 +234,11 @@ private String dependencyNotation(OpenSearchDistribution distribution) {
             extension = distribution.getPlatform() == Platform.WINDOWS ? "zip" : "tar.gz";
 
             if (distroVersion.onOrAfter("1.0.0")) {
-                classifier = ":" + distribution.getPlatform() + "-x64";
+                if (distribution.getArchitecture() == Architecture.ARM64) {
+                    classifier = ":" + distribution.getPlatform() + "-arm64";
+                } else {
+                    classifier = ":" + distribution.getPlatform() + "-x64";
+                }
             } else if (distroVersion.onOrAfter("7.0.0")) {
```

Comment:
Minor: A switch statement would be better. Then we can gracefully fail when a new architecture type is added.

Reply:
good call, updated this.